include/swift/AST/Substitution.h

```diff
@@ -59,6 +59,9 @@ class Substitution {
   /// conformances.
   Substitution subst(ModuleDecl *module,
                      const SubstitutionMap &subMap) const;
+  Substitution subst(ModuleDecl *module,
```

Comment:
In ProtocolConformance::subst we need to hand it off to `Type::gatherAllSubstitutions`, which needs it to resolve substitutions out of BoundGenericTypes.